### PR TITLE
Flink 1.16.1

### PR DESCRIPTION
**Pull request by MartijnVisser**

### CODE REVIEW

File: _posts/2023-01-26-release-1.16.1.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+# Release Notes
+
+<h2>        Bug
+</h2>
+<ul>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-16582'>FLINK-16582</a>] -         NettyBufferPoolTest may have warns on NettyBuffer leak 
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-24119'>FLINK-24119</a>] -         KafkaITCase.testTimestamps fails due to &quot;Topic xxx already exist&quot;
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
test instabilities

File: _posts/2023-01-26-release-1.16.1.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-26037'>FLINK-26037</a>] -         TaskManagerRunner may crash during shutdown sequence
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-26890'>FLINK-26890</a>] -         DynamoDB consumer error consuming partitions close to retention
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-27341'>FLINK-27341</a>] -         TaskManager running together with JobManager are bind to 127.0.0.1
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-27691'>FLINK-27691</a>] -         RankHarnessTest. testUpdateRankWithRowNumberSortKeyDropsToNotLast test failed with result mismatch
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
test stability

File: _posts/2023-01-26-release-1.16.1.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-28102'>FLINK-28102</a>] -         Flink AkkaRpcSystemLoader fails when temporary directory is a symlink
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-28526'>FLINK-28526</a>] -         Fail to lateral join with UDTF from Table with timstamp column
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-28695'>FLINK-28695</a>] -         Fail to send partition request to restarted taskmanager
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-28742'>FLINK-28742</a>] -         Table.to_pandas fails with lit(&quot;xxx&quot;)
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-28766'>FLINK-28766</a>] -         UnalignedCheckpointStressITCase.runStressTest failed with NoSuchFileException
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
test stability

File: _posts/2023-01-26-release-1.16.1.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-28786'>FLINK-28786</a>] -         Cannot run PyFlink 1.16 on MacOS with M1 chip
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-28863'>FLINK-28863</a>] -         Snapshot result of RocksDB native savepoint should have empty shared-state
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-28960'>FLINK-28960</a>] -         Pulsar throws java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlElement
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-28988'>FLINK-28988</a>] -         Incorrect result for filter after temporal join
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29092'>FLINK-29092</a>] -         HadoopPathBasedPartFileWriterTest.testWriteFile failed with AssertionError
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
test stability

File: _posts/2023-01-26-release-1.16.1.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29217'>FLINK-29217</a>] -         CoordinatorEventsToStreamOperatorRecipientExactlyOnceITCase.testConcurrentCheckpoint failed with AssertionFailedError
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
test stability

File: _posts/2023-01-26-release-1.16.1.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29231'>FLINK-29231</a>] -         PyFlink UDAF produces different results in the same sliding window
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29234'>FLINK-29234</a>] -         Dead lock in DefaultLeaderElectionService
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29298'>FLINK-29298</a>] -         LocalBufferPool request buffer from NetworkBufferPool hanging
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29419'>FLINK-29419</a>] -         HybridShuffleITCase.testHybridFullExchangesRestart hangs
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29461'>FLINK-29461</a>] -         ProcessDataStreamStreamingTests.test_process_function unstable
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
test stability

File: _posts/2023-01-26-release-1.16.1.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29677'>FLINK-29677</a>] -         Prevent dropping the current catalog
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29728'>FLINK-29728</a>] -         TablePlanner prevents Flink from starting is working directory is a symbolic link
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29749'>FLINK-29749</a>] -         flink info command support dynamic properties
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29781'>FLINK-29781</a>] -         ChangelogNormalize uses wrong keys after transformation by WatermarkAssignerChangelogNormalizeTransposeRule 
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29788'>FLINK-29788</a>] -         StatefulJobWBroadcastStateMigrationITCase failed in native savepoints
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
test stability

File: _posts/2023-01-26-release-1.16.1.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29803'>FLINK-29803</a>] -         Table API Scala APIs lack proper source jars
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29817'>FLINK-29817</a>] -         Published metadata for apache-flink in pypi are inconsistent and causes poetry to fail
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29827'>FLINK-29827</a>] -         [Connector][AsyncSinkWriter] Checkpointed states block writer from sending records
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29830'>FLINK-29830</a>] -         PulsarSinkITCase$DeliveryGuaranteeTest.writeRecordsToPulsar failed
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
test stability

File: _posts/2023-01-26-release-1.16.1.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29923'>FLINK-29923</a>] -         Hybrid Shuffle may face deadlock when running a task need to execute big size data
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29927'>FLINK-29927</a>] -         AkkaUtils#getAddress may cause memory leak
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-30030'>FLINK-30030</a>] -         Unexpected behavior for overwrite in Hive dialect
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-30133'>FLINK-30133</a>] -         HadoopModuleFactory creates error if the security module cannot be loaded
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-30138'>FLINK-30138</a>] -         BlobServerCleanupTest utilizes failsWithin with a quite low timeout which might result in test instabilities
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
also test-related

File: _posts/2023-01-26-release-1.16.1.md [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+</h2>
+<ul>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-27327'>FLINK-27327</a>] -         Add description about changing max parallelism explicitly leads to state incompatibility
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29134'>FLINK-29134</a>] -         fetch metrics may cause oom(ThreadPool task pile up)
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29155'>FLINK-29155</a>] -         Improve default config of grpcServer in Process Mode
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29198'>FLINK-29198</a>] -         RetryExtension doesn&#39;t make the test fail if the retries are exhausted
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
That's also more like an internal thing that only affects test stability.

File: _posts/2023-01-26-release-1.16.1.md [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-29973'>FLINK-29973</a>] -         connector_artifact should append Flink minor version
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-30291'>FLINK-30291</a>] -         Integrate flink-connector-aws into Flink docs
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-30382'>FLINK-30382</a>] -         Flink 1.16 to integrate KDS/KDF docs from flink-connector-aws
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-30383'>FLINK-30383</a>] -         UseLogicalIdentifier makes datadog consider metric as custom
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-30678'>FLINK-30678</a>] -         NettyConnectionManagerTest.testManualConfiguration appears to be unstable
+</li>
```
Comment (XComp) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
```
test stability

File: q/gradle-quickstart.sh [hunk 12]
```diff
@@ -116,7 +116,7 @@ ext {
     scalaBinaryVersion = '${scalaBinaryVersion}'
     slf4jVersion = '1.7.32'
```
Comment (XComp) [inline q/gradle-quickstart.sh]:
```suggestion
    slf4jVersion = '1.7.36'
```
if we want to fix these in general.

File: q/gradle-quickstart.sh [hunk 13]
```diff
@@ -116,7 +116,7 @@ ext {
...
     log4jVersion = '2.17.1'
-    flinkVersionNew = flinkVersion.toString().replace("-SNAPSHOT", "") >= "1.15"
+    flinkVersionNew = flinkVersion.toString().replace("-SNAPSHOT", "") >= "1.16"
```
Comment (XComp) [inline q/gradle-quickstart.sh:119]: I created FLINK-30768 to improve the version updates in this script.

**Review by XComp — COMMENTED** [review]
I went over the announcement. I'm not sure about the test instabilities. Do we include them? I remember some document where it stated that we should exclude them. But I couldn't find any documentation, anymore.

File: _posts/2023-01-26-release-1.16.1.md [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+# Upgrade Notes
+
+[FLINK-28988 - Incorrect result for filter after temporal join](https://issues.apache.org/jira/browse/FLINK-28988)
+The filter will not be pushed down into both inputs of the event time temporal join.
+This may cause incompatible plan changes compared to Flink 1.16.0, e.g., when the left input is an upsert source 
+(like upsert-kafka connector), the query plan will remove the ChangelogNormalize node in Flink 1.16.1, while it
+did appear in 1.16.0.
+
+[FLINK-29849 - Event time temporal join on an upsert source may produce incorrect execution plan](https://issues.apache.org/jira/browse/FLINK-29849)
+This resolves the correctness issue when do event time temporal join with a versioned table backed by an upsert source.
```
Comment (dawidwys) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
This resolves the correctness issue when doing an event time temporal join with a versioned table backed by an upsert source. 
```

**Review by dawidwys — APPROVED** [review]

**Comment by dawidwys** [comment]
@XComp 

> I'm not sure about the test instabilities. Do we include them? I remember some document where it stated that we should exclude them. But I couldn't find any documentation, anymore.

I believe we should exclude them, as the the first paragraph of the blogpost says 😅 

> Below you will find a list of all bugfixes and improvements (excluding improvements to the build infrastructure and build stability). For a complete list of all changes see:

File: _posts/2023-01-26-release-1.16.1.md [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+[FLINK-28988 - Incorrect result for filter after temporal join](https://issues.apache.org/jira/browse/FLINK-28988)
+The filter will not be pushed down into both inputs of the event time temporal join.
+This may cause incompatible plan changes compared to Flink 1.16.0, e.g., when the left input is an upsert source 
+(like upsert-kafka connector), the query plan will remove the ChangelogNormalize node in Flink 1.16.1, while it
+did appear in 1.16.0.
+
+[FLINK-29849 - Event time temporal join on an upsert source may produce incorrect execution plan](https://issues.apache.org/jira/browse/FLINK-29849)
+This resolves the correctness issue when do event time temporal join with a versioned table backed by an upsert source. 
+When the right input of the join is an upsert source, it no longer generates a ChangelogNormalize node for it.
+This is an incompatible plan change compare to 1.16.0
```
Comment (hlteoh37) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
This is an incompatible plan change compared to 1.16.0
```

**Review by hlteoh37 — APPROVED** [review]

File: _posts/2023-01-26-release-1.16.1.md [hunk 16]
```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+title:  "Apache Flink 1.16.1 Release Announcement"
+date: 2023-01-23T22:00:00.000Z
```
Comment (MartijnVisser) [inline _posts/2023-01-26-release-1.16.1.md]:
```suggestion
date: 2023-01-30T16:00:00.000Z
```

**Pull request merged**